### PR TITLE
[polaris.shopify.com] Add Design Principles page (duplicate)

### LINKED ISSUE #10691: [Markdown] Transfer Design Principles
Edit: Content has been modified for the `Design principles` page.

New name for the page (title and nav): `Pro design language`

Updated content is in this section of the [Google Doc](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit)

[Figma mockup](https://www.figma.com/file/vO0wToyBpYjeIwr8doAAWT/Uplift-polaris.shopify.com?type=design&node-id=529-3880&mode=design&t=BdL7gcZ4aUm8GDZq-4) has been updated.

Images and videos have also been updated: [Google drive
](https://drive.google.com/drive/folders/1zWYVuV3FnbNNtJeQSHxwZRWzGKl9wu3k)

### PR DESCRIPTION
**Pull request by gwyneplaine**
Duplicate of https://github.com/Shopify/polaris/pull/10740 (partially)

### WHY are these changes introduced?

Fixes #10691  

### WHAT is this pull request doing?

Updating the design/index.md page to containg content and layout from the [Design Principles content doc](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit#heading=h.nm44pgerxvao)

28-12-960ff-k4zy0

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/content/design/index.md [hunk 1]
```diff
@@ -1,12 +1,175 @@
...
+          ## Tactility
+
+          **Interfaces incorporate a sense of realness.**
+
+          Primary interaction points provide a dramatic visual response to interaction. Variables like position and lighting change to mimic how physical buttons react when pressed. A hover transition is smooth and unobtrusive, whilst feedback from a click feels more dramatic and visceral.
+
+        </Box>
+      </Grid.Cell>
+      <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 8, xl: 8}}>
+        ![Visual: Animation of buttons being hovered and clicked](/images/design/design-principles-tactility-realness@2x.png)
```
Comment (gwyneplaine) [inline polaris.shopify.com/content/design/index.md:134]: @tjonx  @zecarlostorre 
[The document](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit#heading=h.m850jhu8ex16) specifies the asset as a video, however an image was provided. 

Do we have a requisite animation, or is the doc incorrect?

File: polaris.shopify.com/content/design/index.md [hunk 2]
```diff
@@ -1,12 +1,175 @@
...
+      </Grid.Cell>
+      <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 4, xl: 4}}>
+        <Box padding="400">
+
+          Detailed interactions, such as animating a checkbox when it is ticked, create a sense of familiarity and reference how merchants interact with real-world objects. These animations contribute to making the admin interface feel dynamic and alive, whilst making merchants feel more comfortable and confident in their interactions.
+
+        </Box>
+      </Grid.Cell>
+      <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 8, xl: 8}}>
+      ![Visual: Import, Export, Edit, Print, and other icons that are established](/images/design/design-principles-tactility-animation@2x.png)
```
Comment (gwyneplaine) [inline polaris.shopify.com/content/design/index.md:144]: @tjonx  @zecarlostorre 
[The document](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit#heading=h.m850jhu8ex16) specifies the asset as a video, however an image was provided. 

Do we have a requisite animation, or is the doc incorrect?

**Review by sophschneider — APPROVED** [review]
Tophatted locally, page looks good both on mobile and large screens. Interactions look good.

Didn't review content deeply but seems to match the google doc 👍

File: polaris.shopify.com/content/design/index.md (hunk 2 above)
Comment (zecarlostorre) [reply polaris.shopify.com/content/design/index.md:144]: The images were place holders.
I just put the 3 videos required for this page in here:
https://drive.google.com/drive/folders/1zWYVuV3FnbNNtJeQSHxwZRWzGKl9wu3k?usp=drive_link

Can you make sure they loop?

**Comment by gwyneplaine** [comment]
Closed in favor of #10740

**Pull request closed without merging**